nebula_tests/hesai/hesai_ros_decoder_test.cpp

```diff
@@ -310,12 +294,20 @@ void HesaiRosDecoderTest::ReadBag(
         "Found data in topic " << bag_message->topic_name << ": " << bag_message->time_stamp);
 
       auto extracted_msg_ptr = std::make_shared<pandar_msgs::msg::PandarScan>(extracted_msg);
-      auto pointcloud_ts = driver_ptr_->ConvertScanToPointcloud(extracted_msg_ptr);
-      auto scan_timestamp = std::get<1>(pointcloud_ts);
-      pointcloud = std::get<0>(pointcloud_ts);
 
-      scan_callback(bag_message->time_stamp, scan_timestamp, pointcloud);
-      pointcloud.reset(new nebula::drivers::NebulaPointCloud);
+      for (auto & pkt : extracted_msg_ptr->packets) {
+        auto pointcloud_ts = driver_ptr_->ParseCloudPacket(std::vector<uint8_t>(pkt.data.begin(), std::next(pkt.data.begin(), pkt.size)));
+        auto pointcloud = std::get<0>(pointcloud_ts);
+        auto scan_timestamp = std::get<1>(pointcloud_ts);
+
+        if (!pointcloud) {
+          continue;
+        }
+
+        std::cerr << "PC size: " << pointcloud->size() << std::endl;
```

Comment:
Pointcloud size

Reply:
Addressed in [65be49b](https://github.com/tier4/nebula/pull/127/commits/65be49b1bf31d24ca4a66ab6aab1591df86ff0e9)